.github/workflows/check-paper-integrity.yml

```
@@ -0,0 +1,48 @@
+name: Test Paper Architecture integrity
+on: [pull_request]
+jobs:
+  check:
+    if: github.repository == 'software-mansion/react-native-gesture-handler'
+    runs-on: ubuntu-latest
+    concurrency:
+      group: kotlin-lint-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Use Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'oracle'
+          java-version: '17'
+
+      - name: Use Node.js 18
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18
+          cache: 'yarn'
+
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install node dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Install node dependencies of FabricExample
+        run: (cd FabricExample && yarn install --frozen-lockfile)
+
+      - name: Restore build from cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+            android/build
+            android/.gradle
+          key: ${{ runner.os }}-kotlin-lint-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', 'android/build.gradle') }}
+          
+      - name: Check check integrity
```

```suggestion
      - name: Check old arch integrity
```

I think that one check is enough